Shell/EqualityProxy.cpp

```diff
@@ -261,14 +261,10 @@ Clause* EqualityProxy::apply(Clause* cl)
 
   ASS(_defUnit);
 
-  Clause* res = new(clen) Clause(clen, 
+  auto res = Clause::fromStack(*resLits, 
     NonspecificInference2(InferenceRule::EQUALITY_PROXY_REPLACEMENT, cl, _defUnit));
+  // TODO isn't this done automatically?
```

Comment:
Right? @quickbeam123 might know why this is.

Reply:
Yeah I wan't sure about this. There was quite a couple of places where there was manual assigning of split sets, sometimes with comments that this "should be redundant except for in preproceing" or things like that. I think it would be nice if we could get this uniform somehow (i.e. that it's being done automatically everywhere). 
Do you think that's possible @quickbeam123 ? I only had a glimps on the relevant places in the codebase so I might be overlooking some complications.

Reply:
Look into `Inference.hpp` lines 750-780.

We have different rules for SimplyingInference and for GeneratingInferences regarding the propagation of age. 

We have no rules NonspecificInferences:
```
  /** No special propagation, no extra checks. Use sparingly. */
 ``` 
 because I wasn't able to find a good general that would cover all the cases.

Reply:
I vaguely remember there was no common denominator of all the strange cases and I didn't want to change behavior of any of the inferences I was touching. 

If I remember correctly, for example `GlobalSubsumption` needs a custom call to `setAge` anyway...

Reply:
Hmm, the enum in the constructor solution could have a disabled default, forcing everything to think about age when using nonspecific. Is that what you meant?

Reply:
Yess exactly :)

Reply:
That's a good idea for a new PR. Thanks!